create_temp_dir does not honor permissions and group

arungrover opened the pull request:
<!--- Please review your changes in preview mode -->
<!--- Provide a general summary of your changes in the Title above -->

#### Describe Bug or Feature
create_temp_dir creates a temporary directory but does not honor the permission bits given as input.

#### Describe Your Change
Change in #1098 changed the way a temporary directory used to be created. But this change missed checking for mode.


#### Link to Design Doc
NA

#### Attach Test and Valgrind Logs/Output
I'm going to run a regression. PR is marked as draft.
[test_after.txt](https://github.com/openpbs/openpbs/files/6105460/test_after.txt)
[test_before.txt](https://github.com/openpbs/openpbs/files/6105461/test_before.txt)

[test harness - test_dsh.txt](https://github.com/openpbs/openpbs/files/6233159/test_dsh.txt)

<!--- Pull Request Guidelines: [Pull Request Guidelines](https://pbspro.atlassian.net/wiki/spaces/DG/pages/1187348483/Pull+Request+Guidelines) -->

arungrover commented:
PR is ready for review

bhroam commented on test/tests/selftest/pbs_dshutils_tests.py:
```diff
@@ -86,3 +89,91 @@ def test_pbs_env(self):
+            self.assertEqual(ret['rc'], 0)
+            res = ret['out'][0]
+            u, g, m = [int(v) for v in
+                       res.replace('(', '').replace(')', '').split(', ')]
```
Since you're the one doing the print, and you're just going to get rid of the (), why print them in the first place?

bhroam reviewed (approved):
Just one minor comment that you can choose to address or not.

agrawalravi90 commented on test/fw/ptl/utils/pbs_dshutils.py:
```diff
@@ -2069,55 +2071,66 @@ def create_temp_dir(self, hostname=None, suffix='', prefix='PtlPbs',
+                self.run_copy(hostname, src=tmpdir, dest=tmpdir,
+                              recursive=True, uid=uid, gid=gid,
+                              level=level)
+                self.chmod(hostname, path=tmpdir, mode=mode, runas=asuser)
```
regardless of whether asuser was provided or not, don't we want to do chmod if 'mode' was provided? I know this is not your code, but I'm just wondering if the earlier logic for setting was mode was also incorrect, shouldn't it be outside the if asuser block?

agrawalravi90 reviewed (changes requested):
Minor comments

arungrover commented on test/fw/ptl/utils/pbs_dshutils.py:
```diff
@@ -1561,6 +1561,8 @@ def chown(self, hostname=None, path=None, uid=None, gid=None, sudo=False,
                            runas=runas, level=level)
         if ret['rc'] == 0:
             if gid is not None:
+                if runas is None:
+                    runas = _u
```
chown has already happened before control reaches here. This check it to set runas as the user who owns the file so that it can also change its group.

agrawalravi90 commented:
@arungrover just making sure nothing is pending on my end for this

arungrover commented:
@bhroam  and @agrawalravi90  I've addressed your review comments.
[test_dshutils.txt](https://github.com/openpbs/openpbs/files/6418657/test_dshutils.txt)

agrawalravi90 commented on test/fw/ptl/utils/pbs_dshutils.py:
```diff
@@ -2114,14 +2112,15 @@ def create_temp_dir(self, hostname=None, suffix='', prefix='PtlPbs',
                 # copy temp dir created on localhost to remote as current user
                 self.run_copy(hostname, src=tmpdir, dest=tmpdir,
-                              level=level, preserve_permission=True,
+                              level=level, preserve_permission=False,
```
just wondering, if we are going to do a chmod anyways for either case, can we get rid of the if-else block and just call run_copy with preserve_permission=False? chmod should take care of fixing the permissions right?

agrawalravi90 reviewed (changes requested):
just one small comment, and can you please run a fresh regression with your latest changes?

arungrover commented:
Hey, @agrawalravi90 I've addressed your review comment. I had to rebase the change to do internal testing. I'll post results soon. Please have a look.
[test_dshutils.txt](https://github.com/openpbs/openpbs/files/6429563/test_dshutils.txt)

agrawalravi90 reviewed (commented):
Thanks Arun, changes look fine to me, I'll approve and merge once you post the regression results.

arungrover commented:
Thanks @agrawalravi90 for suggesting me to run regression. Turns out I missed setting preserve_permission to false in one case and a test also needed a change.
I've fixed that in latest commit, please have a look.

Here are the test results - 
Description: Rerun All Tests From #7441
Platforms: UBUNTU1804,SUSE15,UBUNTU1604,CENTOS8,RHEL8,CENTOS7,RHEL7,SLES12,SLES15
Profile: regression
|ID|TOTAL|FAIL|ERROR|TIMEDOUT|SKIP|PASS|
|--|--|--|--|--|--|--|
|7467|4112|1|0|0|1|4110|

One test failure above it not related to this change.
